Mathlib/CategoryTheory/Opposites.lean

```diff
@@ -44,7 +44,7 @@ theorem Quiver.Hom.unop_inj {X Y : Cᵒᵖ} :
   fun _ _ H => congr_arg Quiver.Hom.op H
 #align quiver.hom.unop_inj Quiver.Hom.unop_inj
 
-@[simp]
+@[simp, nolint simpVarHead]
 theorem Quiver.Hom.unop_op {X Y : C} (f : X ⟶ Y) : f.op.unop = f :=
   rfl
```

Comment:
Doesn't this mean that this `@[simp]` lemma is tried on every single term?

Reply:
Yeah. I thought we could just remove this from the `simp` tree but that broke things. I wanted to first make that performance wasn't severely impacted and this slipped off my radar. 

I'll take another look today

Reply:
Another point: I've never seen `simpVarHead` flagged without `by simp` sufficing. This one also!

Reply:
I don't understand your last two comments, sorry.